server/src/main/java/org/opensearch/search/profile/query/InternalProfileComponent.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.profile.query;
+
+import java.util.Collection;
+
+public interface InternalProfileComponent {
```

Comment:
What is the purpose of introducing this interface? I see that only one class implements it now. Is there plan to have multiple implementations of it?

Reply:
Yes, it is extensively used in https://github.com/opensearch-project/OpenSearch/pull/1500 to include `CollectorManager` instances into profiling